### PR TITLE
Update nix files for monorepo

### PR DESCRIPTION
**Pull request by saibatizoku**
* Moves and updates from `./src/jormungandr/flake.nix` to `./flake.nix`
* Removes `flake.lock` in `./src` subdirectories
* Moves `default.nix` and `shell.nix` to top-level, removes duplicates in `./src` subdirectories
* adds crates to the Cargo workspace
* adds `proptest` to `[workspace.depedencies]`
* adds fixes and updates to get nix build to work

### CODE REVIEW

**Review by Mr-Leshiy — APPROVED** [review]
LGTM, but I would wait for approval someone from the devops team

**Comment by minikin** [comment]
@garbas Can you review this, pls?

**Review by cong-or — APPROVED** [review]

**Comment by saibatizoku** [comment]
Thank you @garbas ! I'll address your comments in this PR.

**Pull request merged**
